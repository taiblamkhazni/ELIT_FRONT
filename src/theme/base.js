/**
 * @file base.js
 * @brief Ce fichier d√©finit le composants de style base.
 */
export const base = {
    breakpoints: ["768px"],
    font: "'Roboto sans', sans-serif",
    backgrounds: {
        light: "#E5E5E5",
        gris: "#F9F9F9",
        white: "#FFFFFF",
    },
    fontWeights: {
        light: "300",
        regular: "400",
        bold: "700",
        black: "900",
    },
    fontSizes: {
        Kilo: "32px",
        Hecto: "24px",
        Deca: "20px",
        Base: "18px",
        Deci: "16px",
        Centi: "14px",
    },
    lineHeights: {
        Kilo: "54px",
        Hecto: "48px",
        Deca: "40px",
        Base: "40px",
        Deci: "32px",
        Centi: "24px",
    },
    colors: {
        primaires: {
            blueLight: "#F7FBFF",
            blue: "#116E9C",
            blueDark: "#1F1A28",
            white: "#FFFFFF",
        },
        secondaires: {
            violet: "#7B83E2",
            vertBlue: "#39B7BF",
            blue: "#6FC1E2",
            orange: "#FFAC5F",
            vert: "#DFE997",
            rose: "#FA927D",
            jaune: "#FFFBA0",
            gris: "#F9F9F9",
            grisMedium: "#CCCCCC",
            grisLight: "#E9E9E9",
            grisDark: "#7A7A7A",
        },
        avertissements: {
            succes: "#10B581",
            danger: "#C91432",
        },
    },
}
