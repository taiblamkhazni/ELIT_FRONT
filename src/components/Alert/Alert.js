/**
 * @file Alert.js
 * @brief This module exports Error Warning component
 */
import styled from "styled-components"

/**
 * @brief ErrorAlert : error warning component
 */
export const ErrorAlert = styled.span`
  margin: ${(props) => (props.margin ? props.margin : "initial")};
  color: ${({ theme }) => theme.colors.avertissements.danger};
  text-align: ${(props) => (props.textAlign ? props.textAlign : "initial")};
  font-size: ${(props) => (props.fontSize ? props.fontSize : "0.75rem")};
  font-style: ${(props) => (props.fontStyle ? props.fontStyle : "normal")};
  font-weight: ${(props) => (props.fontWeight ? props.fontWeight : "400")};
`;
