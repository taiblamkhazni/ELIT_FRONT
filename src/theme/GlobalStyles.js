/**
 * @file GloblaStyles.js
 * @brief Ce fichier définit plusieurs composants de style.
 * @details Chaque composant est un style spécifique.
 */
import { createGlobalStyle, css } from "styled-components"
/**
 *
 * @param theme - The theme to use.
 * @param customColor - The color choosed.
 *
 * @returns Composant.
 */
export const scrollBarStyle = css`
  ::-webkit-scrollbar {
    width: 6px;
    background-color: #f5f5f5;
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.1);
    background-color: ${({ theme, customColor }) => customColor ? customColor : theme.colors.primaires.blue};
  }
  scroll-behavior: smooth;
`
/**
 * Determines the color for a project's status button.
 *
 * @param theme - The theme to use.
 *
 * @returns Composant.
 */
export const GlobalStyles = createGlobalStyle`
    body {
        background: ${({ theme }) => theme.backgrounds.white};
        color: ${({ theme }) => theme.colors.primaires.blueDark};
        font-family: ${({ theme }) => theme.font};
        transition: all 0.50s linear;
        box-sizing: border-box;
        margin: 0;
        padding: 0 !important;
        ${scrollBarStyle}
    }
    .ant-modal-wrap {
        overflow-x: hidden;
        ${scrollBarStyle}
    }

    input[type="checkbox"] {
      width: 1.15em;
      height: 1.15em;
      border-radius: 0.15em;
      transform: translateY(-0.075em);
      vertical-align: middle;
      accent-color: ${({ theme }) => theme.colors.primaires.blue};
    }
`
