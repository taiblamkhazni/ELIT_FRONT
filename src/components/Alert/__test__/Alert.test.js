/**
 * @file Alert.test.js
 * @brief Ce fichier contient des tests pour les composant  ErrorAlert.
 */
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import { base } from "theme/base"

import { ErrorAlert } from "../Alert"

import "jest-styled-components"
/**
 * @brief test unitaire pour l'objet Alert.
 */
test("Alert component works", () => {
    const tree = renderer
        .create(
            <ThemeProvider theme={base}>
                <ErrorAlert />
            </ThemeProvider>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule("color", expect.stringContaining("#C91432"))
    expect(tree).toHaveStyleRule("font-style", "normal")
})
