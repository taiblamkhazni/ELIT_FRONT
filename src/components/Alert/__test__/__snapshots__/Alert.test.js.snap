// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component works 1`] = `
.c0 {
  margin: initial;
  color: #C91432;
  text-align: initial;
  font-size: 0.75rem;
  font-style: normal;
  font-weight: 400;
}

<span
  className="c0"
/>
`;
